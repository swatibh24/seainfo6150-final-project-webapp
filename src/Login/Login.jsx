import React, {useEffect, useState} from "react";
import styles from "./Login.module.css";
import { Link } from "react-router-dom";
import Nav from "../Nav/Nav.jsx";

const Login = () => {
  const [submittedFrm, setSubmittedFrm] = useState();
  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedFrm(data);
  };
  return (
    <div>{
        submittedFrm ?(<p className={styles.welcome}>Welcome {submittedFrm.get("username")}!</p>): (
          <form onSubmit={onSubmit} className={styles.contentBox}>
            <div className="logoBox">
            <img className={styles.logo} src="http://shoppopbox.com/wp-content/uploads/2017/09/homechef-logo-square-1.jpg" alt="loginpage" />
            </div>
            <h1 className={styles.loginTitle}>Login</h1>
            <div className={styles.usernameBox}>
              <input type="text" className={styles.input} name="username" id="username" required="required" placeholder="Please Enter Username"/>
            </div>
            <div className={styles.passwordBox}>
              <input type="password" className={styles.input} name="password" id="password" required="required" placeholder="Please Enter Password"/>
            </div>
            <div className={styles.grid}><Link to="/register" className={styles.signUp}>SignUp</Link>
            <input className={styles.login} type="submit" value="Login" />
            </div>
          </form>
        )
      }
    </div>
  )
}

export default Login;