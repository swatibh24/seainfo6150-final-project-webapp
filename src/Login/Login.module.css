.loginContainer {
    font-size: 20px;
    padding: 20px;
    margin-right: auto;
}

.contentBox{
    position: relative;
    z-index: 1;
    max-width: 360px;
    max-height: 500px;
    margin: 100px auto;
    display: block;
    width: 50%;
    padding: 40px 30px 10px 30px;
    background-image: linear-gradient(to right, #06719b, #5d80cc);
}

.logo{
    border-radius: 25%;
    width: 50px;
    height: 50px;
    display:block;
    margin:0 auto;
  }

.logoBox{
    text-align:center;
}

.usernameBox{
    margin-top:20px;
}
.passwordBox{
    margin-top:20px;
}

.loginTitle{
    color:white;
    text-align:center;
}
.alertBox{
    margin-top:20px;
    margin-bottom:20px;
}

.input{
    background-color: #F3E5F5;
    border-radius: 50px !important;
    padding: 12px 15px 12px 15px !important;
    width: 100%;
    box-sizing: border-box;
    border: none !important;
    border: 1px solid #F3E5F5 !important;
    font-size: 16px !important;
    color: #000 !important;
    font-weight: 400;
    margin-bottom:20px;
}

.registerLink{
    font-size: 20px;
    text-decoration: underline;
}
.grid{
  display:flex;
}
.signUp{
    border-color: transparent;
    font-size:1rem;
    width:60%;
    cursor:pointer;
    margin-bottom: 50px;
    outline: none;
    text-decoration: none;
    background-color:white;
    margin-top: 10px;
    font-weight: 600;
    padding: 1rem 2rem;
    text-transform: uppercase;
    transition-property: all;
    transition-duration: 0.5s;
    box-shadow: 1px 1px 4px 0 rgba(0,0,0,.4);
    transition-timing-function: cubic-bezier(0.65,-0.25,0.25,1.95);
    border-radius: 25px;
}
.login{
    border-color: transparent;
    font-size:1rem;
    width:60%;
    margin-left:25%;
    cursor:pointer;
    margin-bottom: 50px;
    outline: none;
    text-decoration: none;
    margin-top: 10px;
    font-weight: 600;
    padding: 1rem 2rem;
    text-transform: uppercase;
    transition-property: all;
    transition-duration: 0.5s;
    box-shadow: 1px 1px 4px 0 rgba(0,0,0,.4);
    transition-timing-function: cubic-bezier(0.65,-0.25,0.25,1.95);
    border-radius: 25px;
}

.alert{
    font-weight: bold;
    margin: 100px 37%;
    color: orange
}
.welcome{
    margin: 15% 25%;
    font-size: 55px;
    font-weight: bold;
    color: grey;
    text-align:center;
}