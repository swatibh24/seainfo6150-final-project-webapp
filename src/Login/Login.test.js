import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Login from "./Login.jsx";
import { BrowserRouter, Link } from "react-router-dom";

it("renders the Login component correctly", () => {
  const { container } = render(<BrowserRouter><Login /></BrowserRouter>);
  expect(container).toMatchSnapshot();
});