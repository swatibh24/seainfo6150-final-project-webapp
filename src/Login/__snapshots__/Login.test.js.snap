// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Login component correctly 1`] = `
<div>
  <div>
    <form
      class="contentBox"
    >
      <div
        class="logoBox"
      >
        <img
          alt="loginpage"
          class="logo"
          src="http://shoppopbox.com/wp-content/uploads/2017/09/homechef-logo-square-1.jpg"
        />
      </div>
      <h1
        class="loginTitle"
      >
        Login
      </h1>
      <div
        class="usernameBox"
      >
        <input
          class="input"
          id="username"
          name="username"
          placeholder="Please Enter Username"
          required=""
          type="text"
        />
      </div>
      <div
        class="passwordBox"
      >
        <input
          class="input"
          id="password"
          name="password"
          placeholder="Please Enter Password"
          required=""
          type="password"
        />
      </div>
      <div
        class="grid"
      >
        <a
          class="signUp"
          href="/register"
        >
          SignUp
        </a>
        <input
          class="login"
          type="submit"
          value="Login"
        />
      </div>
    </form>
  </div>
</div>
`;
